protocols/kad/src/behaviour.rs

```diff
@@ -1903,29 +1892,8 @@ where
             self.address_failed(peer_id, addr);
         }
 
-        // When a connection is established, we don't know yet whether the
-        // remote supports the configured protocol name. Only once a connection
-        // handler reports [`HandlerEvent::ProtocolConfirmed`] do we
-        // update the local routing table.
-
         // Peer's first connection.
         if other_established == 0 {
-            // Queue events for sending pending RPCs to the connected peer.
-            // There can be only one pending RPC for a particular peer and query per definition.
-            for (peer_id, event) in self.queries.iter_mut().filter_map(|q| {
-                q.inner
-                    .pending_rpcs
-                    .iter()
-                    .position(|(p, _)| p == &peer_id)
-                    .map(|p| q.inner.pending_rpcs.remove(p))
-            }) {
-                self.queued_events.push_back(ToSwarm::NotifyHandler {
-                    peer_id,
-                    event,
-                    handler: NotifyHandler::Any,
-                });
-            }
-
             self.connected_peers.insert(peer_id);
```

Comment:
`connected_peers` is a hash set so inserting something repeatedly does not harm. I think we can remove the check for `other_established` and just always do this as part of `preload_new_handler` like you had it before :)

Reply:
To be honest, I don't even understand why we have two data structures here, `connections` and `connected_peers`. It looks like we shouldn't need `connected_peers` because we have `connections`.

However, exploring that would be something for a different PR.

Reply:
> `connected_peers` is a hash set so inserting something repeatedly does not harm. I think we can remove the check for `other_established` and just always do this as part of `preload_new_handler` like you had it before :)

https://github.com/libp2p/rust-libp2p/blob/2d58cbfca20f515af01f92df2b0f8fb5914e649a/protocols/kad/src/behaviour.rs#L2131-L2147

If we put `connected_peers`  as part of `preload_new_handler` will break this debug assert when running `behaviour::test::network_behaviour_on_address_change` test. Is this assertion essential here?

Reply:
Good discovery! Let's keep it as is then :)